src/core/src/bound_evaluate.cpp

```diff
@@ -514,9 +508,11 @@ bool ov::tensor_is_non_negative(const Tensor& bound) {
 bool ov::tensor_has_max_value(const Tensor& bound) {
     const auto bound_constant =
         std::make_shared<op::v0::Constant>(bound.get_element_type(), bound.get_shape(), bound.data());
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    auto max_constant = ngraph::get_constant_max_of_type(bound.get_element_type());
-    OPENVINO_SUPPRESS_DEPRECATED_END
+
+    const auto max_values = ov::util::make_tensor_of_max_value(bound.get_element_type());
+    if (!max_values)
```

Comment:
I think this check is not required. Previous version used nullptr Constant which will make exception or crash when used for evaluation.
Here we use empty tensor which will make exception when accessing its data.
If is not possible to created tensor of maximum value for given precision it doesn't mean input tensor don't have it.

Reply:
Done